security/pkg/pki/ra/k8s_ra_test.go

```diff
@@ -78,27 +79,27 @@ func TestK8sSignWithMeshConfig(t *testing.T) {
 		{
 			name:                  "Root cert is specified in mesh config and Root cert from cert chain is empty(only one leaf cert)",
 			rootCertForMeshConfig: path.Join(env.IstioSrc, "samples/certs", "root-cert.pem"),
-			certChain:             path.Join(env.IstioSrc, "samples/certs", "cert-chain.pem"),
+			certChain:             path.Join(env.IstioSrc, "samples/certs", "leaf-workload-foo-cert.pem"),
```

Comment:
With the update to only include the leaf certificate in the in certChain rather than use `cert-chain.pem` which is a CA, this test is now failing with the following error:

```bash
root cert from signed cert-chain is invalid failed to verify certificate: x509: certificate signed by unknown authority
```

Reply:
put root cert + intermediate ca cert into one pem and used it for root certs to verify the leaf cert, similar like below: 
`cp "$certchain"  "$FINAL_DIR/workload-$sa-root-certs.pem"
cat "$rootcert" >> "$FINAL_DIR/workload-$sa-root-certs.pem"
`
name:                  "Root cert is specified in mesh config and Root cert from cert chain is empty(only one leaf cert)",
			rootCertForMeshConfig: path.Join(env.IstioSrc, "samples/certs", "worklaod-foo-root-certs.pem"),
			certChain:             path.Join(env.IstioSrc, "samples/certs", "leaf-workload-foo-cert.pem")
`

Reply:
Is this a valid scenario? What external ca configuration would this be used for?

Reply:
Basically what we need to test here is to make sure the root cert comming from mesh config and the leaf workload only cert comes from signed cert chain.   The original failure reason for your TC is you have not make the intermediat CA in the root certs.  So to make this work,  make the root cert + intermediate ca cert as root certs in meshconfig.

Reply:
I believe that scenario is covered by the subsequent test `Root cert is specified in mesh config and cert chain contains only intermediate CA(only leaf cert + intermediate CA) `.

Reply:
Okay, I added a test with the root and intermediate CA specified in the MeshConfig and leaf in the cert chain. I also kept the test with just the root in the MeshConfig and the leaf in the cert chain.